packages/arb-token-bridge-ui/src/hooks/useETHPrice.ts

```diff
@@ -1,39 +1,36 @@
-import { useState, useEffect, useCallback, useMemo } from 'react'
+import { useCallback, useMemo } from 'react'
 import axios from 'axios'
-
-export type ETHPrice = 'loading' | 'error' | number
+import useSWR from 'swr'
 
 export type UseETHPriceResult = {
-  ethPrice: ETHPrice
-  toUSD: (etherValue: number) => number
+  ethPrice: number
+  ethToUSD: (etherValue: number) => number
+  error: Error
 }
 
 export function useETHPrice(): UseETHPriceResult {
-  const [ethPrice, setETHPrice] = useState<ETHPrice>('loading')
-
-  useEffect(() => {
-    async function fetchETHPrice() {
-      try {
-        const response = await axios.get(
-          'https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd'
-        )
-
-        setETHPrice(response.data.ethereum.usd)
-      } catch (error) {
-        setETHPrice('error')
-      }
+  const { data, error } = useSWR(
+    'https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd',
+    url => axios.get(url).then(res => res.data),
+    {
+      refreshInterval: 30_000,
+      shouldRetryOnError: true,
+      errorRetryCount: 2,
+      errorRetryInterval: 3_000
     }
+  )
 
-    fetchETHPrice()
-  }, [])
-
-  const toUSD = useCallback(
+  const ethToUSD = useCallback(
     (etherValue: number) => {
+      const ethPrice = data?.ethereum?.usd
       const safeETHPrice = typeof ethPrice === 'number' ? ethPrice : 0
       return etherValue * safeETHPrice
     },
-    [ethPrice]
+    [data]
   )
 
-  return useMemo(() => ({ ethPrice, toUSD }), [ethPrice, toUSD])
+  return useMemo(
+    () => ({ ethPrice: data?.ethereum?.usd, ethToUSD, error }),
+    [data, error, ethToUSD]
+  )
```

Comment:
Small thing: Maybe we should return everything that `useSWR` returns instead of just `data` and `error`, as we may need it at some point, e.g. `isValidating`.

```suggestion
  return { ...swrResult, ethToUSD }
```